### DISCUSSION TITLE
Undefined symbol GOMP_parallel

### DISCUSSION THREAD

**Discussion by marcovc** [coin-or/Ipopt#522] [Q&A] [answered]
Hi,

I've built IpOpt with pardiso. I got the following error message when trying to use it (through SCIP):

Exception of type: DYNAMIC_LIBRARY_FAILURE in file "../../src/Common/IpLibraryLoader.cpp" at line 67:
Exception message: /usr/local/lib/libpardiso600-GNU800-X86-64.so: undefined symbol: GOMP_parallel                                                                          
[nlpi_ipopt.cpp:1167] ERROR: Ipopt returned with application return status -12

It is a bid odd that doing:
 
# ldd /usr/local/lib/libpardiso600-GNU800-X86-64.so
       linux-vdso.so.1 (0x00007ffd7c59d000)
       libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f73f298b000)
       libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f73f283c000)
       libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f73f264a000)
       /lib64/ld-linux-x86-64.so.2 (0x00007f73f2f19000)

seems to show there is no undefined symbols (right?)

Any hints?

Thanks!

**Comment by svigerske** [comment, marked as answer]
`ldd` is not about symbols, but shows you the library dependencies.

Apparently there is an unresolved OpenMP symbol in the Pardiso library, but the pardiso lib misses to list the OpenMP lib (libgomp.so) as a dependency. @oschenk ?

Making Ipopt depend on libgomp may work around that, so try calling Ipopt's configure with `LDFLAGS=-fopenmp`. Or run SCIP with `LD_PRELOAD=libgomp.so` in your environment.

**Comment by marcovc** [comment]
Thanks, I was able to unstall HSL instead and avoid that rabbit hole.
